xml/schema/Core/UserJob.xml

```diff
@@ -0,0 +1,173 @@
+  <add>5.50</add>
+  <log>false</log>
+  <field>
+    <name>id</name>
+    <title>User Job ID</title>
+    <type>int unsigned</type>
+    <required>true</required>
+    <comment>Job ID</comment>
+    <html>
+      <type>Number</type>
+    </html>
+    <add>5.50</add>
+  </field>
+  <primaryKey>
+    <name>id</name>
+    <autoincrement>true</autoincrement>
+  </primaryKey>
+  <field>
+    <name>name</name>
+    <title>User job name</title>
+    <type>varchar</type>
+    <length>64</length>
+    <comment>Unique name for job.</comment>
+    <add>5.50</add>
+  </field>
+  <index>
+    <name>UI_name</name>
+    <fieldName>name</fieldName>
+    <unique>true</unique>
+    <add>5.50</add>
+  </index>
+  <field>
+    <name>created_id</name>
+    <type>int unsigned</type>
+    <title>Created By Contact ID</title>
+    <comment>FK to contact table.</comment>
+    <html>
+      <label>Created By</label>
+    </html>
+    <add>5.50</add>
+  </field>
+  <foreignKey>
+    <name>created_id</name>
+    <table>civicrm_contact</table>
+    <key>id</key>
+    <add>5.50</add>
+    <onDelete>SET NULL</onDelete>
+  </foreignKey>
+  <field>
+    <name>created_date</name>
+    <type>timestamp</type>
+    <default>CURRENT_TIMESTAMP</default>
+    <required>true</required>
+    <title>Import Job Created Date</title>
+    <readonly>true</readonly>
+    <comment>Date and time this job was created.</comment>
+    <add>5.50</add>
+    <html>
+      <type>Select Date</type>
+      <formatType>activityDateTime</formatType>
+    </html>
+  </field>
+  <field>
+    <name>start_date</name>
+    <type>timestamp</type>
+    <required>false</required>
+    <title>Import Job Started Date</title>
+    <comment>Date and time this import job started.</comment>
+    <add>5.50</add>
+    <html>
+      <type>Select Date</type>
+      <formatType>activityDateTime</formatType>
+    </html>
+    <readonly>true</readonly>
+  </field>
+  <field>
+    <name>end_date</name>
+    <type>timestamp</type>
+    <required>false</required>
+    <title>Job Ended Date</title>
+    <comment>Date and time this import job ended.</comment>
+    <add>5.50</add>
+    <html>
+      <type>Select Date</type>
+      <formatType>activityDateTime</formatType>
+    </html>
+    <readonly>true</readonly>
+  </field>
+  <field>
+    <name>expires_date</name>
+    <type>timestamp</type>
+    <required>false</required>
+    <title>Import Job Expires Date</title>
+    <comment>Date and time to clean up after this import job (temp table deletion date).</comment>
+    <add>5.50</add>
+    <html>
+      <type>Select Date</type>
+      <formatType>activityDateTime</formatType>
+    </html>
+  </field>
+  <field>
+    <name>table_name</name>
+    <type>varchar</type>
+    <length>255</length>
+    <comment>Name of the table used (field used for imports and to facilitate temp table cleanup).</comment>
+    <html>
+      <label>Table Name</label>
+    </html>
+    <serialize>COMMA</serialize>
```

Comment:
There would be multiple table names?

Reply:
@totten not for the import case - but per my last comment - when I thought about using it for scheduling reports then the answer *might* be yes - in which case it would be good if the cleanup could clean up more than 1

Reply:
Right, I should've finished reading first. :) 

So... the idea here seems to be: "a big job creates various resources (queues/tables/files/connections/etc), and when it finishes, we should cleanup those resources." The scoping of the table affects the way you think about attached resources. My gut reaction:

* For a limited scope (`civicrm_import_job`), the list of resources will be more predictable. It seems more agreeable to use simple/concrete model ("a spreadsheet importer needs one staging table and one task queue" => `ADD COLUMN table_name VARCHAR, ADD COLUMN queue_id INT`).
* For a generic scope (`civicrm_user_job`), the list of resources will be more variable+diverse. It seems more agreeable to use generic techniques (`ADD COLUMN metadata TEXT`, `hook_userJob($job, $oldStatus, $newStatus) `). The hook-listeners can use `$job->type`, `$job->metadata`, `$newStatus` to decide which resources to cleanup.

    ```php
    Civi::dispatcher()->addListener('hook_civicrm_userJob', 'cleanup');
    function cleanup($e) {
      if ($e->newStatus === 'completed') {
        $tables = $e->job['metadata']['tempTables'] ?? [];
        Schema::dropTables(preg_grep('/-durable-table-pattern-/', $tables));
      }
    });
    ```

(Note: A trade-off to be aware of is that `COMMA` fields and `JSON` fields give weak FKs and can be trickier to query. But I can't think of any specific problems/blockers for the kinds of use-cases/deployments we'd expect.)

Anyway, that's not a hard block your proposed schema. But I would encourage taking a moment to think about that  first.

Reply:
@totten so my take on your take is that hte advantage of having a 'table_name' field disappears if we are not sure where there will be more than one - I think I'm inclined just to drop the field then. I'd rather add it back if we decide we want it than have to drop it later on if we don't